classes/session/SessionDAO.inc.php

```diff
@@ -169,6 +169,22 @@ function sessionExistsById($sessionId) {
 		$row = $result->current();
 		return $row ? (boolean) $row->row_count : false;
 	}
+
+	/**
+     * Delete given user's all sessions except for the given session id
+     *
+     * @param int       $userId
+     * @param string    $sessionId
+     *
+     * @return void
+     */
+    public function deleteAllUserSessionsExceptGivenId($userId, $sessionId) {
+
+        $this->update(
+            'DELETE FROM sessions WHERE user_id = ? AND session_id != ?',
```

Comment:
I see you've just followed existing patterns here, but this is an aspect where we're adopting Laravel's query builder toolset as we write/rewrite SQL queries. Eventually we'd like to remove any old-style `DAO::update` calls, which used to be executed using ADODB and are now around for backwards compatibility. (See the deprecations in `lib/pkp/classes/db/DAO.inc.php`.)

(For the `main` branch, we're also adding type hinting to functions where possible.)

Note that `!=` in SQL is probably not going to work with PostgreSQL (thought I haven't checked). The "most" standard SQL operator is `<>`.

Reply:
Yes, I was trying to follow the same code pattern so that it does not get polluted with different code style and adaptations . But I agree using the Laravel style is much better option as in that case we do not have to worry about the SQL compatibility for different DB drivers . I will update it to Laravel style `Query Builder` format as I can see `Eloquent` has not yet adopted into the system .

Reply:
about the `!=` and `<>`, as I have checked, up to `PostgreSQL 14` (which is the latest one) has support for both (https://www.postgresql.org/docs/14/functions-comparison.html) .